src/examples/AddDataExample.java

```diff
@@ -0,0 +1,194 @@
+
+package net.opentsdb.examples;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+import com.stumbleupon.async.Callback;
+import com.stumbleupon.async.Deferred;
+
+import net.opentsdb.core.TSDB;
+import net.opentsdb.core.WritableDataPoints;
+import net.opentsdb.uid.NoSuchUniqueName;
+import net.opentsdb.uid.UniqueId.UniqueIdType;
+import net.opentsdb.utils.Config;
+
+/**
+ * Examples for how to add points to the tsdb.
+ * 
+ * @author Bryan Hernandez
+ * 
+ */
+public class AddDataExample {
+
+  public static void main(String[] args) throws Exception {
+
+    // Set these as arguments so you don't have to keep path information in
+    // source files
+    String pathToConfigFile = args[0]; // e.g.
+                                       // "/User/thisUser/opentsdb/src/opentsdb.conf"
+
+    // Create a config object with a path to the file for parsing. Or manually
+    // override settings.
+    // e.g. config.overrideConfig("tsd.storage.hbase.zk_quorum", "localhost");
+    Config config = new Config(pathToConfigFile);
+    final TSDB tsdb = new TSDB(config);
+
+    // Declare new metric
+    String metricName = "dummyFromjavaAPI";
+    // First check to see it doesn't already exist
+    byte[] byteMetricUID; // we don't actually need this for the first
+                          // .addPoint() call below.
+    // TODO: Ideally we could just call a not-yet-implemented tsdb.uIdExists()
+    // function.
+    try {
+      byteMetricUID = tsdb.getUID(UniqueIdType.METRIC, metricName);
+    } catch (IllegalArgumentException iae) {
+      System.out.println("Metric name not valid.");
+      iae.printStackTrace();
+    } catch (NoSuchUniqueName nsune) {
+      // If not, great. Create it.
+      byteMetricUID = tsdb.assignUid("metric", metricName);
+    }
+
+    // Make a single datum
+    long timestamp = System.currentTimeMillis();
+    long value = 314159;
+    // Make key-val
+    Map<String, String> tags = new HashMap<String, String>(1);
+    tags.put("dummy-key", "dummy-val1");
+
+    // This is an optional errorback to handle when there is a failure.
+    class errBack implements Callback<String, Exception> {
+      public String call(final Exception e) throws Exception {
+        String message = ">>>>>>>>>>>Failure!>>>>>>>>>>>";
+        System.out.println(message);
+        return message;
+      }
+    }
+    ;
+
+    // Start timer
+    long startTime1 = System.nanoTime();
+    // Write datum to tsdb
+    Deferred<Object> deferred = tsdb.addPoint(metricName, timestamp, value,
+        tags);
+
+    // Add the callbacks to the deferred object. (They might have already
+    // returned, btw)
+    // This will cause the calling thread to wait until the add has completed.
+    deferred.addErrback(new errBack());
+
+    // End timer.
+    long elapsedTime1 = (System.nanoTime() - startTime1) / (1000 * 1000);
+    System.out.println("\nAdding 1 point took: " + elapsedTime1
```

Comment:
You should also add a success callback and put the timer print out there. Then join on the deferred before moving on to the next step.